sunpy/io/file_tools.py

```diff
@@ -144,18 +145,15 @@ def write_file(fname, data, header, filetype='auto', **kwargs):
     * This routine currently only supports saving a single HDU.
     """
     if filetype == 'auto':
-        if not isinstance(fname, str):
-            raise ValueError("Can not automatically detect filetype for non-string fname argument")
```

Comment:
I've removed this error, because anything that can't be coerced into a Path will throw an error at the `pathlib.Path` stage.